### DISCUSSION TITLE
Sudden jumps in daily energy

### DISCUSSION THREAD

**Discussion by MarkoPaasila** [emporia-vue-local/esphome#118] [General]
This is a water heater. The rounded corners seem to correspond to reality, but the straight up edges certainly don't:
![image](https://user-images.githubusercontent.com/6001014/217468743-dc1b2d51-ad66-448d-a029-00d23012a867.png)

Power for this device is reported correctly. Spikes in total energy coincide with power usage. It seems like powerouts (maybe reboots) also cause a spike in daily totals - as you can see at 08:18. This doesn't happen on my other esphome devices with total_daily_energy.

**Comment by flaviut** [comment]
Looks normal, here's mine:

![2023-02-08-071046_970x938_scrot](https://user-images.githubusercontent.com/5213469/217526148-29017aa0-4ea2-48bd-aacf-038ba77bb98f.png)

**Reply by MarkoPaasila** [reply]
Yours looks normal indeed; 3kW for 20mins = 1kWh. With 6kW and 20mins I get 40kWh.

**Reply by flaviut** [reply]
Oh! I didn't notice the scale there, my bad! That's very odd. Looks like you're getting the normal ramp up, but always right after it spikes.

[Looks like someone else ran into this too](https://github.com/esphome/issues/issues/2730#issuecomment-973878572), can you try their solution of [setting `heartbeat: 30s` in the power sensor's filters](https://esphome.io/components/sensor/index.html#heartbeat)?

**Reply by MarkoPaasila** [reply]
Indeed, I had a `delta: 5` added as an extra filter which would have messed with the energy integration.

**Comment by clowrey** [comment]
Ooooh! I just had this same issue with another ESPhome project using the "[Total Daily Energy Sensor](https://esphome.io/components/sensor/total_daily_energy.html?highlight=total+daily)"
It looks like the easiest best way to fix it is to use "method: left" modifier in the sensor setup! 
The other way of doing these KWh totals is using the Home Assistant "helpers" "Add Riemann sum integral sensor" And the same issue will occur there if you don't choose "left" as the type.. It should be the default IMO but its not.

**Reply by flaviut** [reply]
This is the better solution ^

Would recommend this over the `heartbeat: 30s`. I'll update the sample config later.

**Reply by MarkoPaasila** [reply]
I think I’ll use both, since I use `delta` too. Otherwise I’ll have a related problem on the other side of the power surge.

**Reply by flaviut** [reply]
> Otherwise I’ll have a related problem on the other side of the power surge.

~Probably not, because there's noise in the system when it is on that means that the reading changes slightly much more frequently than every 30s.~

Never mind, didn't catch the `delta` part 

But regardless, it definitely won't cause any problems.

**Comment by MarkoPaasila** [comment]
Thanks guys for helping me sort this out. Funny how small innocent things like `delta` can have seemingly unrelated effects :-)
